How to create custom highcharts file, errors encountered.

Following the documentation for [how to create custom highcharts file](https://www.highcharts.com/docs/getting-started/how-to-create-custom-highcharts-files).

#### Expected behaviour

At this step:

Open the unpacked folder highcharts-master in a command line or terminal and run **npm install** to install any required dependencies. 

I expect no errors.

#### Actual behaviour

However, on Windows 10, I get many errors. I've tried executing NPM install from VS Code's Bash terminal and Powershell terminal. I've also tried it from the Windows command prompt. **The package installation fails completely**.

Here are samples of the errors:

```
C:\wwwroot\highcharts10\highcharts-master>npm install
npm WARN deprecated topo@3.0.3: This module has moved and is now available at @hapi/topo. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated hoek@4.2.1: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated hoek@5.0.4: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated lolex@6.0.0: lolex has been renamed to @sinonjs/fake-timers. No API changes made. Please use the new package instead
```

And some more:

```
npm WARN deprecated joi@13.7.0: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated istanbul@0.4.5: This module is no longer maintained, try this instead:
npm WARN deprecated   npm i nyc
npm WARN deprecated Visit https://istanbul.js.org/integrations for other alternatives.
npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
npm ERR! code 1
npm ERR! path C:\wwwroot\highcharts10\highcharts-master\node_modules\node-expat
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node-gyp rebuild
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.0.0
npm ERR! gyp info using node@16.10.0 | win32 | x64
npm ERR! gyp ERR! find Python
npm ERR! gyp ERR! find Python Python is not set from command line or npm configuration
npm ERR! gyp ERR! find Python Python is not set from environment variable PYTHON
```


#### Live demo with steps to reproduce
<!-- template: https://jsfiddle.net/highcharts/LLExL/ -->

N/A.

#### Product version
<!--- Highcharts, Highstock or Highmaps plus version number -->

Highcharts v10.2



#### Affected browser(s)

N/A.


**Other points, questions:**


1. Is the package install not supported on Windows?
2. Or are there other prerequisites? 

I also deleted the package.lock file and tried the install again. Same results.

Some more details. On a hunch, I removed the following lines from package.json's dev dependencies:


```
    "@highcharts/highcharts-documentation-generators": "github:highcharts/highcharts-documentation-generators#v0.5.33",
    "@highcharts/highcharts-utils": "github:highcharts/highcharts-utils#v1.0.1",
    "highcharts-assembler": "github:highcharts/highcharts-assembler#v1.4.1",
    "highcharts-declarations-generator": "github:highcharts/highcharts-declarations-generator#v1.1.20",
    "highsoft-websearch": "github:highcharts/highsoft-websearch#v0.0.3"
```

Running NPM installed, installed the other dev dependencies with only some minor warnings about deprecated packages.

But "@highcharts/map-collection": "^2.0.0" failed to install:

```
> highcharts@10.2.0 prepare
> husky install

fatal: not a git repository (or any of the parent directories): .git
```


**Are there rights issues involved to access these Highcharts repositories?**

After this, running **NPM run test**, produces these results:

```
> highcharts@10.2.0 test
> npx gulp test-ts --force && npx gulp test --force --speak

Error: Cannot find module 'highcharts-assembler/src/build.js'

```

Thanks, @pawellysy for your very prompt reply.

I'm running stable, slightly newer versions of Node and NPM.

$ npm --version
8.11.0

$ node --version
v16.16.0

I had seen that error about Python missing, but I took the documentation literally

`Open the unpacked folder highcharts-master in a command line or terminal and run npm install to install any **required dependencies**`

so I didn't attempt to install it earlier.  I've now **installed Python**, and it's on the Windows path:

MINGW64 /c/wwwroot/highcharts10/highcharts-master
$ py --version
Python 3.10.5

I deleted the prior highcharts-master folder. I restarted Windows and then cleaned out the temp folder for good measure.

Opened a Bash shell in highcharts-master and ran NPM install.

I still get below errors (this time, this is the complete output). Hopefully, the problem isn't with the newer versions of Node/NPM from what you are using.

(see my note after these errors)

```
MINGW64 /c/wwwroot/highcharts10/highcharts-master
$ npm install
npm WARN deprecated topo@3.0.3: This module has moved and is now available at @hapi/topo. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated hoek@4.2.1: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated hoek@5.0.4: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated lolex@6.0.0: lolex has been renamed to @sinonjs/fake-timers. No API changes made. Please use the new package instead
npm WARN deprecated hoek@6.1.3: This module has moved and is now available at @hapi/hoek. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated joi@13.7.0: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated istanbul@0.4.5: This module is no longer maintained, try this instead:
npm WARN deprecated   npm i nyc
npm WARN deprecated Visit https://istanbul.js.org/integrations for other alternatives.
npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
npm ERR! code 1
npm ERR! path C:\wwwroot\highcharts10\highcharts-master\node_modules\node-expat
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node-gyp rebuild
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.0.0
npm ERR! gyp info using node@16.16.0 | win32 | x64
npm ERR! gyp info find Python using Python version 3.10.5 found at "C:\Program Files\Python310\python.exe"
npm ERR! gyp http GET https://nodejs.org/download/release/v16.16.0/node-v16.16.0-headers.tar.gz
npm ERR! gyp http 200 https://nodejs.org/download/release/v16.16.0/node-v16.16.0-headers.tar.gz
npm ERR! gyp http GET https://nodejs.org/download/release/v16.16.0/SHASUMS256.txt
npm ERR! gyp http GET https://nodejs.org/download/release/v16.16.0/win-x86/node.lib
npm ERR! gyp http GET https://nodejs.org/download/release/v16.16.0/win-x64/node.lib
npm ERR! gyp http GET https://nodejs.org/download/release/v16.16.0/win-arm64/node.lib
npm ERR! gyp http 200 https://nodejs.org/download/release/v16.16.0/SHASUMS256.txt
npm ERR! gyp http 404 https://nodejs.org/download/release/v16.16.0/win-arm64/node.lib
npm ERR! gyp http 200 https://nodejs.org/download/release/v16.16.0/win-x64/node.lib
npm ERR! gyp http 200 https://nodejs.org/download/release/v16.16.0/win-x86/node.lib
npm ERR! gyp ERR! find VS
npm ERR! gyp ERR! find VS msvs_version not set from command line or npm config
npm ERR! gyp ERR! find VS VCINSTALLDIR not set, not running in VS Command Prompt
npm ERR! gyp ERR! find VS could not use PowerShell to find Visual Studio 2017 or newer, try re-running with '--loglevel silly' for more details
npm ERR! gyp ERR! find VS looking for Visual Studio 2015
npm ERR! gyp ERR! find VS - not found
npm ERR! gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
npm ERR! gyp ERR! find VS
npm ERR! gyp ERR! find VS **************************************************************
npm ERR! gyp ERR! find VS You need to install the latest version of Visual Studio
npm ERR! gyp ERR! find VS including the "Desktop development with C++" workload.
npm ERR! gyp ERR! find VS For more information consult the documentation at:
npm ERR! gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
npm ERR! gyp ERR! find VS **************************************************************
npm ERR! gyp ERR! find VS
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: Could not find any Visual Studio installation to use
npm ERR! gyp ERR! stack     at VisualStudioFinder.fail (C:\Users\kthom\AppData\Roaming\nvm\v16.16.0\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:122:47)
npm ERR! gyp ERR! stack     at C:\Users\kthom\AppData\Roaming\nvm\v16.16.0\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:75:16
npm ERR! gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (C:\Users\kthom\AppData\Roaming\nvm\v16.16.0\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:363:14)
npm ERR! gyp ERR! stack     at C:\Users\kthom\AppData\Roaming\nvm\v16.16.0\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:71:14
npm ERR! gyp ERR! stack     at C:\Users\kthom\AppData\Roaming\nvm\v16.16.0\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:384:16
npm ERR! gyp ERR! stack     at C:\Users\kthom\AppData\Roaming\nvm\v16.16.0\node_modules\npm\node_modules\node-gyp\lib\util.js:54:7
npm ERR! gyp ERR! stack     at C:\Users\kthom\AppData\Roaming\nvm\v16.16.0\node_modules\npm\node_modules\node-gyp\lib\util.js:33:16
npm ERR! gyp ERR! stack     at ChildProcess.exithandler (node:child_process:406:5)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:527:28)
npm ERR! gyp ERR! stack     at maybeClose (node:internal/child_process:1092:16)
npm ERR! gyp ERR! System Windows_NT 10.0.18363
npm ERR! gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\kthom\\AppData\\Roaming\\nvm\\v16.16.0\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd C:\wwwroot\highcharts10\highcharts-master\node_modules\node-expat
npm ERR! gyp ERR! node -v v16.16.0
npm ERR! gyp ERR! node-gyp -v v9.0.0
npm ERR! gyp ERR! not ok

```

I'm just seeing this:

**npm ERR! gyp ERR! find VS You need to install the latest version of Visual Studio**

I see there is a free version called [Visual Studio Community](https://visualstudio.microsoft.com/vs/community/#:~:text=09%2D08%3A00-,Visual%20Studio%20Community,web%20applications%20and%20cloud%20services.). I'm going to save this comment so as not to lose the above documentation. I'll download VS, install, reboot. And we'll see what happens.

Visual Studio fixed the errors. I still get the errors about deprecated packages.

To confirm the package installation, I ran the below NPM scripts. 

`test`
`ts-compile:test`

Both reported some warnings.

Ts-compile:test reported this error:

```
C:\wwwroot\highcharts10\highcharts-master>npm run ts-compile:test

> highcharts@10.2.0 ts-compile:test
> npx tsc -p test && npx tsc -p samples

npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
test/test-controller.ts:35:42 - error TS2503: Cannot find namespace 'Highcharts'.
```

So I'm not sure yet if I'm going to be able to build minified ES6 modules successfully or not. But I'll give it a go over the next day or so.

--- a/docs/doc-config.js
+++ b/docs/doc-config.js
@@ -22,6 +22,7 @@ module.exports = {
         { from: 'maps/create-custom-maps-for-highmaps', to: 'maps/create-custom-maps' },
         { from: 'stock/understanding-highstock', to: 'stock/understanding-highcharts-stock' },
         { from: 'getting-started/compatibility', to: 'getting-started/system-requirements' },
+        { from: 'getting-started/how-to-create-custom-highcharts-files', to: 'getting-started/how-to-create-custom-highcharts-packages' },
         { from: 'chart-and-series-types/packed-bubble-charts', to: 'chart-and-series-types/packed-bubble' },
         { from: 'chart-concepts/navigator', to: 'stock/navigator' },
         { from: 'chart-concepts/range-selector', to: 'stock/range-selector' },

--- a/docs/getting-started/how-to-create-custom-highcharts-packages.md
+++ b/docs/getting-started/how-to-create-custom-highcharts-packages.md
@@ -1,9 +1,9 @@
-Creating custom Highcharts files
-================================
+Creating custom Highcharts packages
+===================================
 
-Thanks to ES modules you can create your own custom Highcharts package files. A
-benefit of using a custom file can be optimization of browser load speed due to
-lower filesize and less files to request.
+Thanks to ES modules you can create your own custom Highcharts package.
+A benefit of using a custom file can be optimization of browser load speed due
+to lower filesize and less files to request.
 
 Follow the steps below to get started.
 
@@ -16,17 +16,19 @@ Go to the [`highcharts/highcharts`](https://github.com/highcharts/highcharts)
 repository and click on "Code" and select "Download ZIP". Once the zip file is
 downloaded, unpack it to a desired location.
 
-Open the unpacked folder `highcharts-master` in a command line or terminal and
-run `npm install` to install any required dependencies. On Macs this step
-requires the Installation of Xcode app or the Command Line Tools, which you run
-with `xcode-select --install`.
+The following steps require Node.js, which you can download and install from the
+[Node.js website](https://nodejs.org/en/). Highcharts supports the TSL variant.
+
+When Node.js is installed, open you command line, shell, or terminal, and go to
+the unpacked folder `highcharts-master`. There you have to run `npm install` to
+install the required dependencies for building custom Highcharts files.
 
 
 
 Create a custom master file
 ---------------------------
 
-Go in your editor to the unpacked folder `highcharts-master/ts/masters` and
+Go in your editor to the unpacked folder `highcharts-master/ts/masters/` and
 create a new file, named for example `custom.src.ts`. In this example we want a
 basic line chart. To achieve this we will need a setup that lists all the needed
 ES modules like this:
@@ -51,35 +53,41 @@ Please note that the order of the imported ES modules sometimes have to match
 the dependencies of each file. Optional additions therefore should come last.
 See other masters for order information.
 
-For similar examples take a look at the other master files.
+For similar examples take a look at existing master files.
 
 
 
 Create the custom package file
 ------------------------------
 
-Run `npx gulp scripts` to build all package files out of the master files.
-Optional run `npx gulp scripts-compile` in addition to get minified versions.
+Run `npx gulp scripts --force` to build all package files out of the master
+files. Optional run `npx gulp scripts-compile` in addition to get minified
+versions.
 
-In our example the new file `ts/masters/custom.src.ts`
-becomes the new package file `code/custom.src.js`.
+In our example the new file `ts/masters/custom.src.ts` becomes the new package
+file `code/custom.src.js`. The minified version can be found as
+`code/custom.js`. You can pick these package files from the `code/` folder and
+use it in your project.
 
-You can pick the package file `code/custom.src.js` and use it in your project.
-
+For ESM-compatible files you have to copy the `code/es-modules/` folder. Feel
+free to rename the folder for your needs. This folder might have a large size,
+but ESM will pick only necessary files from it for your project. A bundler like
+`parcel` or `webpack` can help in the final stage of your project to optimize
+the loading time further.
 
 
 Use your custom package file
 ----------------------------
 
 According to our [installation](./installation.md) guide you can reference your
-custom package file either as an ES module...
+custom package file either as an ES6 module...
 
 ```html
 <html>
     <body>
         <div id="container"></div>
         <script type="module">
-            import Highcharts from 'custom.src.js';
+            import Highcharts from './es-modules/masters/custom.src.js';
             Highcharts.chart('container', {
                 series: [{
                     type: 'line',
@@ -111,4 +119,3 @@ custom package file either as an ES module...
     </body>
 </html>
 ```
-

--- a/docs/getting-started/how-to-create-custom-highcharts-packages.md
+++ b/docs/getting-started/how-to-create-custom-highcharts-packages.md
@@ -28,7 +28,7 @@ install the required dependencies for building custom Highcharts files.
 Create a custom master file
 ---------------------------
 
-Go in your editor to the unpacked folder `highcharts-master/ts/masters/` and
+In your editor, go to the unpacked folder `highcharts-master/ts/masters/` and
 create a new file, named for example `custom.src.ts`. In this example we want a
 basic line chart. To achieve this we will need a setup that lists all the needed
 ES modules like this:
@@ -53,7 +53,7 @@ Please note that the order of the imported ES modules sometimes have to match
 the dependencies of each file. Optional additions therefore should come last.
 See other masters for order information.
 
-For similar examples take a look at existing master files.
+For similar examples take a look at the existing master files.
 
 
 

--- a/docs/sidebars.js
+++ b/docs/sidebars.js
@@ -10,7 +10,7 @@ module.exports = {
             'getting-started/your-first-chart',
             'getting-started/how-to-set-options',
             'getting-started/frequently-asked-questions',
-            'getting-started/how-to-create-custom-highcharts-files',
+            'getting-started/how-to-create-custom-highcharts-packages',
             'getting-started/optional-dependencies'
         ],
         'Chart concepts': [

